routes/web.php

```diff
@@ -1,71 +1,80 @@
-    Route::get('users', [UserController::class, 'index'])->name('users.index');
+    Route::controller(BlessingController::class)->prefix('blessings')->name('blessings.')->group(function () {
+        Route::get('/', 'index')->name('index');
+        Route::get('/create', 'create')->name('create');
+        Route::post('/', 'store')->name('store');
+    });
 });
 
+// Authenticated Routes
 Route::middleware(['auth'])->group(function () {
-    Route::get('users/{user}', [UserController::class, 'show'])->name('users.show');
-    Route::get('users/{user}/edit', [UserController::class, 'edit'])->name('users.edit');
-});
+    Route::controller(GameController::class)->prefix('games')->name('games.')->group(function () {
+        Route::get('/', 'index')->name('index');
+        Route::get('/{game}', 'show')->name('show');
+    });
 
-Route::middleware(['auth', 'admin'])->group(function () {
-    Route::get('blessings', [BlessingController::class, 'index'])->name('blessings.index');
-    Route::get('blessings/create', [BlessingController::class, 'create'])->name('blessings.create');
-    Route::post('blessings', [BlessingController::class, 'store'])->name('blessings.store');
-});
+    Route::controller(CharacterController::class)->prefix('games/{game}/characters/{character}')->name('characters.')->group(function () {
+        Route::get('/', 'show')->name('show');
+        Route::get('/claim', 'claim')->name('claim');
+        Route::get('/target', 'target')->name('target');
+        Route::post('/attack', 'attack')->name('attack');
+        Route::get('/upgrade', 'upgrade')->name('upgrade');
+        Route::post('/unlock-element', 'unlockElement')->name('element.store');
+        Route::post('/unlock-armor', 'unlockArmor')->name('armor.store');
+        Route::post('/unlock-weapon', 'unlockWeapon')->name('weapon.store');
+        Route::post('/unlock-special', 'unlockSpecial')->name('unlockSpecial'); // Consider renaming to specials.store?
+        Route::post('/heal-heart', 'healHeart')->name('healHeart'); // Consider renaming to hearts.store or similar?
+    });
 
-Route::middleware(['auth'])->group(function () {
-    Route::get('blessings/{blessing}', [BlessingController::class, 'withCharacter'])->name('blessings.withCharacter');
-    Route::get('games/{game}/characters/{character}/blessings/{blessing}', [BlessingController::class, 'show'])->name('blessings.show');
-    Route::post('games/{game}/characters/{character}/blessings/{blessing}/claim', [BlessingController::class, 'claim'])->name('blessings.claim');
+    Route::controller(UserController::class)->prefix('users')->name('users.')->group(function () {
+        Route::get('/{user}', 'show')->name('show');
+        Route::get('/{user}/edit', 'edit')->name('edit');
+    });
+
+    Route::controller(BlessingController::class)->name('blessings.')->group(function () {
+        Route::get('blessings/{blessing}', 'withCharacter')->name('withCharacter'); // Does this need character context?
```

Comment:
No, people are following this link from an NFC chip so they will not have the character context. So I use it as a public url that will redirect to the show url with the game and character context if the person who scanned the nfc chip is a player.